File: gems/decomplex/examples/kotlin/weighted-inlined-complexity.kt
fun checkout(user: User, cart: Cart) { validate_user(user); apply_discount(cart); process_payment(user, cart); audit_cart(cart) }
fun validate_user(user: User): Boolean { if (user.active() && !user.suspended()) { if (user.profile.complete()) { return true } else { return false } } else { return false } }
fun apply_discount(cart: Cart): Int { if (cart.total > 100 && eligible()) { if (holiday()) { return 20 } else if (loyalty_month()) { return 15 } else { return 10 } }; return 0 }
fun process_payment(user: User, cart: Cart) { if (gateway.ready()) { if (cart.total > 0 && user.active()) { if (fraud_check(user)) { charge(user, cart) } else { decline(user) } } } }
fun audit_cart(cart: Cart) { for (item in cart.items) { if (item.taxable()) { if (item.region && item.amount > 0) { record_tax(item) } } } }

File: gems/decomplex/examples/python/weighted-inlined-complexity.py
def checkout(user, cart):
    validate_user(user)
    apply_discount(cart)
    process_payment(user, cart)
    audit_cart(cart)

def validate_user(user):
    if user.active() and not user.suspended():
        if user.profile.complete(): return True
        else: return False
    else: return False

def apply_discount(cart):
    if cart.total > 100 and eligible():
        if holiday(): return 20
        elif loyalty_month(): return 15
        else: return 10
    return 0

def process_payment(user, cart):
    if gateway.ready():
        if cart.total > 0 and user.active():
            if fraud_check(user): charge(user, cart)
            else: decline(user)

def audit_cart(cart):
    for item in cart.items:
        if item.taxable():
            if item.region and item.amount > 0:
                record_tax(item)
